.github/workflows/update-release.yml

```diff
@@ -0,0 +1,58 @@
+name: Check for a new Rust release
+
+on:
+  # Allow for this to be manually run.
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Rust release version'
+        type: string
+        required: true
+
+  # Run at midnight every day.
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  check_for_update:
+    name: Check if there is a new release
+    env:
+      version: ${{inputs.version}}
+    outputs:
+      release_version: ${{ steps.check.outputs.version }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the latest release tag name
+        if: env.version == ''
+        run: echo "version=$(gh release view --repo rust-lang/rust --json tagName --jq '.tagName')" >> "$GITHUB_ENV"
+
+      - uses: actions/checkout@v4
+
+      - name: Check if we already have the latest release
+        id: check
+        run: test -f "recipes-devtools/rust/rust-bin-cross_${version}.bb" || echo "version=${version}" >> "$GITHUB_OUTPUT"
+
+  create_pr:
+    name: Create release PR
+    needs: check_for_update
+    if: needs.check_for_update.outputs.release_version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Generate new files for the release
+        run: ./build-new-version.sh ${{needs.check_for_update.outputs.release_version}}
+
+      - name: Commit new files
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          branch: feature/rust-${{needs.check_for_update.outputs.release_version}}
+          create_branch: true
+          commit_message: "Rust ${{needs.check_for_update.outputs.release_version}}"
+          add_options: '--all'
+          push_options: '--force'
```

Comment:
So I better understand, what's the reasoning for requiring `--all` and `--force`?

Reply:
Sure, `--all` is needed because the files which get generated by `build-new-version.sh` are untracked. I went with this route instead of targeting specific files so that you don't need to modify this if you decide to change the output of `build-new-version.sh` in the future.

I put `--force` so that the branch will get overwritten if you want to run this manually, in case something got mis-committed. IIRC, `--force` isn't required for this to work. It does have a side-effect where if you have a release PR up for more than 24 hours, this workflow will run again and move the head of the branch. The workflow won't create a new PR though due to conflicts. I can remove `--force` if you want.

Here's an example PR: https://github.com/gharveymn/meta-rust-bin/pull/2
You can try running it manually here: https://github.com/gharveymn/meta-rust-bin/actions/workflows/update-release.yml (click "Run workflow" and input the version)